src/shared/components/results/ResultsFacets.tsx

```diff
@@ -1,21 +1,52 @@
 import { FC } from 'react';
-import { Facets, Facet } from 'franklin-sites';
+import { Facets, Facet, Loader } from 'franklin-sites';
 
 import TaxonomyFacet from './TaxonomyFacet';
+import ErrorHandler from '../error-pages/ErrorHandler';
 
 import useNS from '../../hooks/useNS';
-import { mainNamespaces } from '../../types/namespaces';
+import useNSQuery from '../../hooks/useNSQuery';
+import useDataApiWithStale from '../../hooks/useDataApiWithStale';
 
-import { FacetObject } from '../../../uniprotkb/types/responseTypes';
+import { mainNamespaces, Namespace } from '../../types/namespaces';
+import Response from '../../../uniprotkb/types/responseTypes';
 
 import helper from '../../styles/helper.module.scss';
-import './styles/results-view.scss';
 
-const ResultsFacets: FC<{ facets: FacetObject[]; isStale?: boolean }> = ({
-  facets,
-  isStale,
-}) => {
-  const ns = useNS();
+import './styles/results-data.scss';
+
+const ResultsFacets: FC = () => {
+  const namespace = useNS() || Namespace.uniprotkb;
+
+  const { url: initialApiUrl } = useNSQuery({ size: 0, withFacets: true });
+
+  const {
+    data,
+    error,
+    loading,
+    progress,
+    headers,
+    status,
+    isStale,
+  } = useDataApiWithStale<Response['data']>(initialApiUrl);
+
+  if (error || !(loading || data) || !namespace) {
+    return <ErrorHandler status={status} />;
+  }
+
+  const total = headers?.['x-totalrecords']
```

Comment:
If there's 0 results there should be no facets as they are calculated based on the results set. So this could be removed.

Reply:
Is this comment in the right place in the code?

Reply:
L37 - I'm wondering whether the check of `total` L45 is necessary and if not we could simplify the logic a bit.

Reply:
When there are no results the facets still exist:
```
[
  {
    "label": "Status",
    "name": "reviewed",
    "allowMultipleSelection": false
  },
  {
    "label": "Model organisms",
    "name": "model_organism",
    "allowMultipleSelection": true
  },
  {
    "label": "Proteins with",
    "name": "proteins_with",
    "allowMultipleSelection": true
  },
  {
    "label": "Protein Existence",
    "name": "existence",
    "allowMultipleSelection": true
  },
  {
    "label": "Annotation Score",
    "name": "annotation_score",
    "allowMultipleSelection": true
  },
  {
    "label": "Sequence length",
    "name": "length",
    "allowMultipleSelection": true,
    "values": [
      {
        "label": "1 - 200",
        "value": "[1 TO 200]",
        "count": 0
      },
      {
        "label": "201 - 400",
        "value": "[201 TO 400]",
        "count": 0
      },
      {
        "label": "401 - 600",
        "value": "[401 TO 600]",
        "count": 0
      },
      {
        "label": "601 - 800",
        "value": "[601 TO 800]",
        "count": 0
      },
      {
        "label": ">= 801",
        "value": "[801 TO *]",
        "count": 0
      }
    ]
  }
]
```
We could try to interpret this object to decide if there are not facets but I think it's easier to condition on `total` as it's immediately available.

Reply:
Ah of course! 😭